### DISCUSSION TITLE
Measuring coverage of tests using jacoco is not including unit tests

### DISCUSSION THREAD

**Discussion by tlopesPT** [quarkusio/quarkus#25075] [Q&A]
Hey there 👋 ,

I'm trying to get a consolidated test coverage report for unit and quarkus tests using jacoco (to be used by sonarqube).

I've been following this guide [setting-up-jacoco](https://quarkus.io/guides/tests-with-coverage#setting-up-jacoco) and testing with the provided quickstart [tests-with-coverage-quickstart](https://github.com/quarkusio/quarkus-quickstarts/tree/main/tests-with-coverage-quickstart)

So far I've been able to generate the **individual reports** for:
- coverage for @QuarkusTest by adding the quarkus-jacoco dependency
- coverage for unit tests by adding the jacoco-maven-plugin

However, things get tricky when trying to get a consolidated report.
Using both the quarkus-jacoco dependency and the jacoco-maven-plugin will either generate a report with 0% test coverage (perhaps an issue with the append configuration on the maven plugin?) or a report that only reflects the coverage provided by the quarkus tests, ignoring the unit tests.

In order to reproduce this issue, clone the test coverage quickstart project linked above, disable the quarkus test **testGreetingEndpoint()** and run `mvn clean verify`.

By opening the generated report at target/jacoco-report/index.html we can verify that the greeting() method in the GreetingService has 0% coverage, even thought the unit test for this method is enabled (we disabled the quarkus test since it would cover this method and hide the fact that the unit test doesn't), and so it should be at 100%.

- If both quarkus tests are enabled, we get 100% coverage, which is correct
- if the 'testGreetingEndpoint()' quarkus test is disabled, we get 57% coverage, **which is wrong**, this is not considering the coverage provided by the unit test, since we can see that 'greeting(String name)' method has 0% coverage when it should have 100%
- if we disable both quarkus tests, we get 0% coverage (unclear if this is expected, since the docs say you must have at least 1 quarkus test in order to use the quarkus-jacoco dependency)
- if we comment both quarkus tests (instead of disabling), we get 31% coverage, so in this scenario it is actually considering the coverage from the unit test

In conclusion, it seems that unit tests coverage is being ignored in the consolidated report using the provided instructions both in the guide and quickstart (which has an outdated readme regarding the generated reports).

In these screenshots we can see the expected coverage result provided in Intellij and the incorrect report provided by jacoco.

![quickstart-test-disabled](https://user-images.githubusercontent.com/19928394/164516865-283dffb2-1b6a-4046-8631-d5c6788ee4a0.png)
![report-1](https://user-images.githubusercontent.com/19928394/164516911-841ffc4b-4d5b-4512-a058-3723ee927c17.png)
![report-2](https://user-images.githubusercontent.com/19928394/164516919-a85388f9-81b6-4898-8406-8ce4e7c68536.png)

Hopefully you can tell me what I'm missing.

Thanks,
Tiago
